pkg/conn/gorilla/gorilla.go

```diff
@@ -235,25 +243,38 @@ func (ws *WebSocket) write(v interface{}) error {
 }
 
 func (ws *WebSocket) initialize() {
-	go func() {
-		for {
-			select {
-			case <-ws.close:
-				return
-			default:
-				var res rpc.RPCResponse
-				err := ws.read(&res)
-				if err != nil {
-					if errors.Is(err, net.ErrClosed) {
-						break
-					}
-					ws.logger.Error(err.Error())
-					continue
+	for {
+		select {
+		case <-ws.closeChan:
+			return
+		default:
+			var res rpc.RPCResponse
+			err := ws.read(&res)
+			if err != nil {
+				shouldExit := ws.handleError(err)
```

Comment:
The function should exit on **any** error returned from `ws.Conn.ReadMessage`.   You can fix by inlining `WebSocket.read` here and exiting when the call to `ws.Conn.ReadMessage` returns an error.  Because `handleError` does not accurately detect all websocket errors, this thing can loop endlessly and eventually panic (the gorilla detects useless application loops).